### ISSUE TITLE
VOD remote should utilize `proxy_pass` instead of hard-coding upstreams

### ISSUE THREAD

**Issue by benwilber** [kaltura/nginx-vod-module#134]
Example use case:

```
upstream mp4_proxy {
  server localhost:6000;
}

server {
  listen 6000;
  # local dnsmasq for resolving $proxy_host
  resolver 127.0.0.1;

  location ~* /hls/(?<proxy_scheme>https?):/+(?<proxy_host>[^/]+)(?<proxy_path>.+) {
    proxy_pass $proxy_scheme://$proxy_host$proxy_path;
    proxy_set_header Host $proxy_host;
  }
}

server {
  listen 80;
  server_name example.com;

  location /child {
    internal;
    vod_child_request;
  }

  location /hls {
    vod hls;
    vod_mode remote;
    vod_child_request_path /child;
    vod_upstream mp4_proxy;
  }
```

This configuration makes it possible to "HLS-ify" any public `.mp4` file:
```
$ wget -q -O- http://example.com/hls/http://cdn2.streamable.com/video/2ca521e0017111e5a591fb6333e5fc88.mp4/index.m3u8
#EXTM3U
#EXT-X-TARGETDURATION:10
#EXT-X-ALLOW-CACHE:YES
#EXT-X-PLAYLIST-TYPE:VOD
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:1
#EXTINF:5.312,
http://example.com/hls/http://cdn2.streamable.com/video/2ca521e0017111e5a591fb6333e5fc88.mp4/seg-1-v1-a1.ts
#EXT-X-ENDLIST
```

However, it is unnecessarily complicated by the `vod_upstream mp4_proxy` requirement.  I wonder if it should just be allowed to use `proxy_pass` directly, and the user can decide if they want to `proxy_pass` to an `upstream`, or some other server directly.

**Comment by erankor** [comment]
Thanks for your suggestion, I agree there are advantages to using proxy_pass. Actually, when I started working on this module I weighted both options, but ended up writing my own upstream handler since the proxy pass alternative didn't work for me. Since then, and while working on some other module, I learned how to overcome these issues.
All that said, I estimate it will take some amount of work to migrate the implementation into proxy_pass, and I probably won't get to it soon. 
I will close this issue for now, and will add it to my 'future improvements' list, maybe one day I'll get to it... :)
